terraform_files/vsphere/main.tf

```diff
@@ -0,0 +1,93 @@
+provider "vsphere" {
+  user                 = var.vsphere_username
+  password             = var.vsphere_password
+  vsphere_server       = var.vsphere_vcenter
+  allow_unverified_ssl = true
+}
+
+data "vsphere_datacenter" "datacenter" {
+  name = var.vsphere_datacenter
+}
+
+data "vsphere_compute_cluster" "cluster" {
+  name          = var.vsphere_cluster
+  datacenter_id = data.vsphere_datacenter.datacenter.id
+}
+
+data "vsphere_datastore" "datastore" {
+  name          = var.vsphere_datastore
+  datacenter_id = data.vsphere_datacenter.datacenter.id
+}
+
+data "vsphere_network" "network" {
+  name          = var.vsphere_network
+  datacenter_id = data.vsphere_datacenter.datacenter.id
+}
+
+resource "vsphere_tag_category" "category" {
+  name        = "openshift-${var.cluster_name}"
+  description = "Added by openshift-install do not remove"
+  cardinality = "SINGLE"
+
+  associable_types = [
+    "VirtualMachine",
+    "ResourcePool",
+    "Folder"
+  ]
+}
+
+resource "vsphere_tag" "tag" {
+  name        = var.cluster_name
+  category_id = vsphere_tag_category.category.id
+  description = "Added by openshift-install do not remove"
+}
+
+resource "vsphere_folder" "folder" {
+  count = 1
+  path          = var.cluster_name
+  type          = "vm"
+  datacenter_id = data.vsphere_datacenter.datacenter.id
+  tags          = [vsphere_tag.tag.id]
+}
+
+resource "vsphere_file" "ISO_UPLOAD" {
+  datacenter       = var.vsphere_datacenter
+  datastore        = var.vsphere_datastore
+  source_file      = var.iso_download_path
+  destination_file = "ISOs/${basename(var.iso_download_path)}"
+}
+
+resource "vsphere_virtual_machine" "vm" {
+  count = var.masters_count
+
+  name                 = "${var.cluster_name}-master-${count.index}"
+  resource_pool_id     = data.vsphere_compute_cluster.cluster.resource_pool_id
+  datastore_id         = data.vsphere_datastore.datastore.id
+  num_cpus             = var.master_vcpu
+  num_cores_per_socket = var.vsphere_control_plane_cores_per_socket
+  memory               = var.master_memory
+  guest_id             = "coreos64Guest"
+  folder               = var.cluster_name
+  enable_disk_uuid     = "true"
+  wait_for_guest_net_timeout  = "0"
+  wait_for_guest_net_routable = "false"
+
+  network_interface {
+    network_id = data.vsphere_network.network.id
+  }
+  
+  disk {
```

Comment:
Should follow naming like we have for libvirt
https://github.com/openshift/assisted-test-infra/blob/51d5d1c28822dba48ac56dd4210117d0b606a272/terraform_files/baremetal/main.tf#L100-L107